pkg/sysfs/system.go

```diff
@@ -467,9 +467,10 @@ func (sys *system) discoverCPUs() error {
 
 	offline, err := sys.SetCpusOnline(true, nil)
 	if err != nil {
-		return fmt.Errorf("failed to set CPUs online: %v", err)
+		sys.Warn("failed to set CPUs online: %v", err)
+	} else {
+		defer sys.SetCpusOnline(false, offline)
 	}
-	defer sys.SetCpusOnline(false, offline)
```

Comment:
This is an excellent change because the old behavior didn't work well with running tests using synthetic `/sys` data.

Reply:
Would you like me to split that out to a PR of its own. We could merge it alone then before considering the rest of this.